rpc/http.go

```diff
@@ -166,10 +177,23 @@ func (hc *httpConn) doRequest(ctx context.Context, msg interface{}) (io.ReadClos
 	if err != nil {
 		return nil, err
 	}
-	req := hc.req.WithContext(ctx)
-	req.Body = ioutil.NopCloser(bytes.NewReader(body))
+	req, err := http.NewRequestWithContext(ctx, "POST", hc.url, ioutil.NopCloser(bytes.NewReader(body)))
+	if err != nil {
+		return nil, err
+	}
 	req.ContentLength = int64(len(body))
 
+	// set headers
+	hc.mu.Lock()
+	for key, vals := range hc.headers {
+		req.Header.Del(key)
```

Comment:
Why do you have to delete the key before setting the key & Vals even though the request is a fresh new request?

Reply:
Yeah, you're right, I got confused.